src/lib/crypto/sol-message-digest-common.c

```diff
@@ -317,6 +320,8 @@ _sol_message_digest_free(struct sol_message_digest *handle)
     _sol_message_digest_thread_fini(handle);
 
     SOL_VECTOR_FOREACH_IDX (&handle->pending_feed, pf, i) {
+        if (handle->on_feed_done)
+            handle->on_feed_done((void *)handle->data, handle, pf->blob, -ECANCELED);
```

Comment:
hum... but that just accounts what is to be processed. Look into `_sol_message_digest_thread_fini()`, what was processed but not dispatched, will still be in `handle->pending_dispatch`.

This list is consumed from `_sol_message_digest_thread_feedback()`, where the vector is stolen to a local variable and re-initialized to zero (took me few minutes to get that, and I was the one who wrote the code... ouch!)